subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/SubscriptionDescriptorFileValidator.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.justice.maven.generator.io.files.parser;
+
+import static java.lang.String.format;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class SubscriptionDescriptorFileValidator {
+
+    private YamlFileToJsonObjectConverter yamlFileToJsonObjectConverter;
+
+    public SubscriptionDescriptorFileValidator(final YamlFileToJsonObjectConverter yamlFileToJsonObjectConverter) {
+        this.yamlFileToJsonObjectConverter = yamlFileToJsonObjectConverter;
+    }
+
+    public void validate(final Path filePath) {
+        final JSONObject convert;
+        try {
+            convert = yamlFileToJsonObjectConverter.convert(filePath);
+        } catch (IOException ex) {
+            throw new SubscriptionDescriptorException(format("Unable to convert to JSON file %s ", filePath.toString()), ex);
+        }
+        schema().validate(convert);
```

Comment:
can't this and the declaration of convert all just be in the try block?

Reply:
Agreed